crates/client/src/client.rs

```diff
@@ -1620,3 +1725,183 @@ impl FuelClient {
+                    |e| {
+                        io::Error::other(format!(
+                            "Failed to convert RPC block to internal block: {e:?}"
+                        ))
+                    },
+                )
+            }
+            Payload::Remote(remote) => {
+                let RemoteBlockResponse { location } = remote;
+                match location {
+                    Some(Location::S3(s3)) => {
+                        let RemoteS3Bucket {
+                            bucket,
+                            key,
+                            endpoint,
+                            ..
+                        } = s3;
+                        let s3_url = endpoint
+                            .unwrap_or_else(|| "https://s3.amazonaws.com".to_string());
+                        let zipped_bytes =
+                            Self::get_block_from_s3_bucket(&s3_url, &bucket, &key)
+                                .await?;
+                        let block_bytes = Self::unzip_bytes(&zipped_bytes);
+                        let block =
+                            ProtoBlock::decode(block_bytes.as_slice()).map_err(|e| {
+                                io::Error::other(format!("Failed to decode block: {e}"))
+                            })?;
+                        let (block, receipts) = fuel_block_from_protobuf(
+                            block,
+                            &[],
+                            Bytes32::default(),
+                        )
+                        .map_err(|e| {
+                            io::Error::other(format!(
+                                "Failed to convert RPC block to internal block: {e:?}"
+                            ))
+                        })?;
+                        Ok((block, receipts))
+                    }
+                    _ => Err(io::Error::other("Remote blocks are not supported yet")),
+                }
+            }
+        }
+    }
+    async fn get_block_from_s3_bucket(
+        url: &str,
+        bucket: &str,
+        key: &str,
+    ) -> io::Result<prost::bytes::Bytes> {
+        let client = Self::aws_client(url).await;
+        tracing::info!("getting block from bucket: {} with key {}", bucket, key);
+        let req = client.get_object().bucket(bucket).key(key);
+        let obj = req.send().await.map_err(|e| {
+            io::Error::other(format!("Failed to get object from S3: {e:?}"))
+        })?;
+        let bytes = obj
+            .body
+            .collect()
+            .await
+            .map_err(|e| {
+                io::Error::other(format!("Failed to get object from S3: {e:?}"))
+            })?
+            .into_bytes();
+        Ok(bytes)
+    }
+
+    async fn aws_client(url: &str) -> Client {
+        let credentials = DefaultCredentialsChain::builder().build().await;
+        let _aws_region =
+            std::env::var("AWS_REGION").expect("AWS_REGION env var must be set");
```

Comment:
This is not a valid assumption on the client side. Even when it's required, the AWS SDK itself will throw an error when it doesn't have the right data.

Reply:
Thanks. Nice catch. I just moved a bunch of stuff over from the test code but didn't fix all of the differences.